test/functional/aws_test.go

```diff
@@ -0,0 +1,162 @@
+				t.Logf("instance: %v , has multi-AZ disabled, instance ARN: %v ", instance.DBName, instance.DBInstanceArn)
+				return nil, errors.New("error: Multi-AZ is disabled")
+			}
+			if instance.StorageEncrypted != aws.Bool(true) {
+				t.Logf("instance: %v , has database encryption disabled,  instance ARN: %v ", instance.DBName, instance.DBInstanceArn)
+				return nil, errors.New("error: encryption is disabled")
+			}
+			if instance.PreferredBackupWindow == nil {
+				t.Logf("instance: %v , has automated backups disabled,  instance ARN: %v ", instance.DBName, instance.DBInstanceArn)
+				return nil, errors.New("error: automated backups are disabled")
+			}
+			if instance.DeletionProtection != aws.Bool(true) {
+				t.Logf("instance: %v , has deletion protection disabled,  instance ARN: %v ", instance.DBName, instance.DBInstanceArn)
+				return nil, errors.New("error: deletion protection is disabled")
+			}
+			checkedResources[aws.StringValue(instance.DBName)] = true
+
+		}
+	}
+}
+
+func newAWSClients(session *session.Session) *awsClients {
+	return &awsClients{
+		rdsClient:         rds.New(session),
+		elasticacheClient: elasticache.New(session),
+		s3Client:          s3.New(session),
+	}
+}
+
+func setAWSCredEnvar(f *framework.Framework) error {
+	secret := &corev1.Secret{}
+	err := f.Client.Get(goctx.TODO(), types.NamespacedName{Name: awsCredsSecretName, Namespace: awsCredsNamespace}, secret)
+	if err != nil {
+		return fmt.Errorf("error getting secret: %v from cluster: %w ", awsCredsSecretName, err)
+	}
+	if string(secret.Data["aws_access_key_id"]) != "" && string(secret.Data["aws_secret_access_key"]) != "" {
+		os.Setenv("AWS_ACCESS_KEY_ID", string(secret.Data["aws_access_key_id"]))
+		os.Setenv("AWS_SECRET_ACCESS_KEY", string(secret.Data["aws_secret_access_key"]))
+		return nil
+	}
+	return fmt.Errorf("error setting %v envar, secret data may not exist", awsCredsSecretName)
+}
+
+// Wait for status of all resources to become "available", return true if all available
+func awsResourcesAvailable(dbinstances []*rds.DBInstance, cacheClusters []*elasticache.CacheCluster) bool {
+	for _, dbinstance := range dbinstances {
+		if dbinstance.DBInstanceStatus != aws.String(resourceStatusAvailable) {
+			return false
+		}
+	}
+	for _, cacheCluster := range cacheClusters {
+		if cacheCluster.CacheClusterStatus != aws.String(resourceStatusAvailable) {
+			return false
+		}
+	}
+	return true
+}
+
+func getRDSResources(session *session.Session) ([]*rds.DBInstance, error) {
+	client := newAWSClients(session)
+	clusterDescribeInput := &rds.DescribeDBInstancesInput{}
+	clusterDescribeOutput, err := client.rdsClient.DescribeDBInstances(clusterDescribeInput)
+	if err != nil {
+		return nil, fmt.Errorf("failed to describe database clusters : %w", err)
+	}
+	var RDSDBInstances []*rds.DBInstance
+	for _, dbInstance := range clusterDescribeOutput.DBInstances {
+		RDSDBInstances = append(RDSDBInstances, dbInstance)
+	}
+	return RDSDBInstances, nil
```

Comment:
I think we can just return `clusterDescribeOutput.DBInstances` as this is an `[]*rds.DBInstance` we do not need to build it ourselves